prov/rxm/src/rxm_cq.c

```diff
@@ -420,18 +425,19 @@ ssize_t rxm_cq_handle_large_data(struct rxm_rx_buf *rx_buf)
 	rx_buf->hdr.state = RXM_LMT_READ;
 
 	for (i = 0; i < rx_buf->rma_iov->count; i++) {
+		size_t copy_len = MIN(rx_buf->rma_iov->iov[i].len,
+				      total_recv_len);
 		ret = ofi_copy_iov_desc(&iov[0], &desc[0], &count,
 					&rx_buf->recv_entry->rxm_iov.iov[0],
 					&rx_buf->recv_entry->rxm_iov.desc[0],
 					rx_buf->recv_entry->rxm_iov.count,
-					&index, &offset, rx_buf->rma_iov->iov[i].len);
+					&index, &offset, copy_len);
```

Comment:
Since this won't return -FI_ETOOSMALL anymore, we need to detect truncation and write CQ error entry.

Reply:
We detect truncation in `rxm_finish_recv()` and write CQ error entry there.

Reply:
Got it. We can skip checking for `ret` and writing a completion error then. At least for now `ofi_copy_iov_desc` would return only `-FI_ETOOSMALL`. Since we have eliminated that we can just add `assert(!ret)`